### PR TITLE
Add test for the unsupported browser screen

### PR DESCRIPTION
**Pull request by dbkr**
Test a handful of APIs that are easy to disable with an init script and assert that the app display the "Unsupported Browser" screen.

Fixes https://github.com/element-hq/element-web/issues/27734
Requires https://github.com/matrix-org/matrix-react-sdk/pull/12782 (or some other fix to prevent libraries referencing APIs before modernizr)

## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

### CODE REVIEW

**Review by t3chguy — COMMENTED** [review]
How about a screenshot test?

File: playwright/snapshots/app-loading/feature-detection.spec.js/unsupported-browser-linux.png
Comment (t3chguy) [inline playwright/snapshots/app-loading/feature-detection.spec.js/unsupported-browser-linux.png:1]: I don't think this is what we're expecting :D

File: playwright/e2e/app-loading/feature-detection.spec.js
Comment (t3chguy) [inline playwright/e2e/app-loading/feature-detection.spec.js:1]: Shouldn't this file be ts?

File: playwright/snapshots/app-loading/feature-detection.spec.js/unsupported-browser-linux.png
Comment (dbkr) [reply playwright/snapshots/app-loading/feature-detection.spec.js/unsupported-browser-linux.png:1]: well this is fun... seems it's just not working on linux...

**Review by t3chguy — APPROVED** [review]

**Pull request merged**